core/src/main/scala/pureconfig/ConvertHelpers.scala

```diff
@@ -0,0 +1,161 @@
+    string => location =>
+      f(string) match {
+        case Some(t) => Right(t)
+        case None => Left(CannotConvert(string, ct.runtimeClass.getName, "", location, None))
+      }
+
+  @deprecated(message = "The usage of Try has been deprecated. Please use fromStringReader instead", since = "0.6.0")
+  def fromString[T](fromF: String => Try[T]): ConfigConvert[T] = new ConfigConvert[T] {
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, T] = stringToTryConvert(fromF)(config)
+    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
+  }
+
+  def fromStringReader[T](fromF: String => Option[ConfigValueLocation] => Either[ConfigReaderFailure, T]): ConfigConvert[T] = new ConfigConvert[T] {
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, T] = stringToEitherConvert(fromF)(config)
+    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(t)
+  }
+
+  def fromStringReaderTry[T](fromF: String => Try[T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromStringReader[T](tryF(fromF))
+  }
+
+  def fromStringReaderOpt[T](fromF: String => Option[T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromStringReader[T](optF(fromF))
+  }
+
+  @deprecated(message = "The usage of Try has been deprecated. Please use fromNonEmptyStringReader instead", since = "0.6.0")
+  def fromNonEmptyString[T](fromF: String => Try[T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromNonEmptyStringReader[T](fromF andThen tryToEither)
+  }
+
+  def fromNonEmptyStringReader[T](fromF: String => Option[ConfigValueLocation] => Either[ConfigReaderFailure, T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromStringReader(string => location => ensureNonEmpty(ct)(string)(location).right.flatMap(s => fromF(s)(location)))
+  }
+
+  def fromNonEmptyStringReaderTry[T](fromF: String => Try[T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromNonEmptyStringReader[T](tryF(fromF))
+  }
+
+  def fromNonEmptyStringReaderOpt[T](fromF: String => Option[T])(implicit ct: ClassTag[T]): ConfigConvert[T] = {
+    fromNonEmptyStringReader[T](optF(fromF))
+  }
+
+  @deprecated(message = "The usage of Try has been deprecated. Please use fromStringConvert instead", since = "0.6.0")
+  def stringConvert[T](fromF: String => Try[T], toF: T => String): ConfigConvert[T] =
+    fromStringConvert[T](fromF andThen tryToEither, toF)
+
+  def fromStringConvert[T](fromF: String => Option[ConfigValueLocation] => Either[ConfigReaderFailure, T], toF: T => String): ConfigConvert[T] = new ConfigConvert[T] {
+    override def from(config: ConfigValue): Either[ConfigReaderFailures, T] = stringToEitherConvert(fromF)(config)
+    override def to(t: T): ConfigValue = ConfigValueFactory.fromAnyRef(toF(t))
+  }
+
+  def fromStringConvertTry[T](fromF: String => Try[T], toF: T => String)(implicit ct: ClassTag[T]): ConfigConvert[T] = {
```

Comment:
The reason is that the error representation in `Try` and even more in `Option` are very primitive and carry less information compared to `Either[ConfigReaderFailure]`, which means that it can be difficult for the user to convert them into an `Either`. We provide helpers that try to get the best out of those types.